src/Features/ERDDAP/Map/WLPlatformLayer.tsx

```diff
@@ -103,12 +102,27 @@ export const WLPlatformLayer = ({ platform, selected, old = false }: PlatformLay
     setDisplay(display)
   }, [floodThreshold])
 
+  const query = sensorPage
+    ? {
+        start: searchParams.get("start"),
+        end: searchParams.get("end"),
+        datum: searchParams.get("datum"),
+      }
+    : {
+        start: getIsoForPicker(threeDaysAgoRounded()),
+        end: getIsoForPicker(weeksInFuture(1)),
+        datum: "datum_mllw_meters",
+      }
+
```

Comment:
Yes, will refactor!

Reply:
And totally agree about wrapping it up into a function that handles it instead of having to deal with all the different scenarios separately. I'll update after these test run